bot.user.js

```diff
@@ -373,58 +373,100 @@ var bot = (function() {
             return a.distance - b.distance;
         },
 
+        checkCollision2: function(circle, collisionPoint){
+            var collisionCircle = canvas.collisionScreenToCanvas({
+                x: collisionPoint.xx,
+                y: collisionPoint.yy,
+                radius: 20 * collisionPoint.sc * canvas.getScale()
+            });
+            
+            if (canvas.circleIntersect(circle, collisionCircle)) {
+                bot.changeGoalCoords(collisionPoint);
+                return true;
+            }
+
+            return false;
+        },
+
         // Checks to see if you are going to collide with anything in the collision detection radius
-        checkCollision: function(circle) {
-            if (!window.collisionDetection) return false;
+        checkCollision: function(headCircle, bodyCircle) {
+
+            if (!window.collisionDetection)
+                return false;
+
             if (window.visualDebugging) {
-                canvas.drawDot(circle.x, circle.y, circle.radius, 'blue', false);
+                canvas.drawDot(bodyCircle.x, bodyCircle.y, bodyCircle.radius, 'blue', false);
+                canvas.drawDot(headCircle.x, headCircle.y, headCircle.radius, 'yellow', false);
             }
 
-            if (bot.collisionPoints[0] != null) {
-                var collisionCircle = canvas.collisionScreenToCanvas({
-                    x: bot.collisionPoints[0].xx,
-                    y: bot.collisionPoints[0].yy,
-                    radius: 20 * bot.collisionPoints[0].sc * canvas.getScale()
-                });
+            if (bot.collisionPoints.length > 0) {
 
-                if (canvas.circleIntersect(circle, collisionCircle)) {
-                    bot.changeGoalCoords();
+                // first check if anything is in the inner circle
+                if (bot.checkCollision2(bodyCircle, bot.collisionPoints[0]))
                     return true;
-                }
+
+                // then check if a snake head is inside the outer circle.
+                if (bot.closestHeadPoint !== null)
+                    if (bot.checkCollision2(headCircle, bot.closestHeadPoint))
+                        return true;
             }
+
             return false;
         },
 
-        getCollisionPoints: function() {
+        setCollisionPoints: function() {
+            bot.closestHeadPoint = null;
+
+            var closestHeadPointDistance = Number.MAX_VALUE;
             var collisionPoints = [];
-            for (var snake in window.snakes) {
-                if (window.snakes[snake].nk != window.snake.nk) {
-                    for (var pts in window.snakes[snake].pts) {
-                        if (!window.snakes[snake].pts[pts].dying) {
+
+            for (var snakenum in window.snakes) {
```

Comment:
I hadn't heard of this rule before, so I looked it up and found this: http://stackoverflow.com/a/4166584/1628916

On the other hand all you did was rename a variable...

@ErmiyaEskandary do you think the bot is being too aggressive?